From 064acb7882a7d9897dc5f8117ca9a94970387701 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Wed, 10 Nov 2021 15:55:32 +0000
Subject: [PATCH] dtoverlays: Remove i2c0mux and i20if status from
 edt-ft5406.dtsi

edt-ft5406.dtsi is included from vc4-kms-dsi-7inch which was
also setting i2c0mux and i2c0if status fields. This meant that
dtoverlay wouldn't apply the overlay due to multiple fragments
changing the same parameter.

Move the enable from edt-ft5406.dtsi to edt-ft5406-overlay.dts
for when it should be needed as an independent overlay.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 .../arm/boot/dts/overlays/edt-ft5406-overlay.dts | 16 ++++++++++++++++
 arch/arm/boot/dts/overlays/edt-ft5406.dtsi       |  8 --------
 .../dts/overlays/vc4-kms-dsi-7inch-overlay.dts   |  2 +-
 3 files changed, 17 insertions(+), 9 deletions(-)

--- a/arch/arm/boot/dts/overlays/edt-ft5406-overlay.dts
+++ b/arch/arm/boot/dts/overlays/edt-ft5406-overlay.dts
@@ -8,3 +8,19 @@
 /plugin/;
 
 #include "edt-ft5406.dtsi"
+
+/ {
+	fragment@0 {
+		target = <&i2c0if>;
+		__overlay__ {
+			status = "okay";
+		};
+	};
+
+	fragment@1 {
+		target = <&i2c0mux>;
+		__overlay__ {
+			status = "okay";
+		};
+	};
+};
--- a/arch/arm/boot/dts/overlays/edt-ft5406.dtsi
+++ b/arch/arm/boot/dts/overlays/edt-ft5406.dtsi
@@ -27,7 +27,6 @@
 		__overlay__ {
 			#address-cells = <1>;
 			#size-cells = <0>;
-			status = "okay";
 			ft5406: ts@38 {
 				compatible = "edt,edt-ft5406";
 				reg = <0x38>;
@@ -38,13 +37,6 @@
 		};
 	};
 
-	fragment@13 {
-		target = <&i2c0if>;
-		__overlay__ {
-			status = "okay";
-		};
-	};
-
 	__overrides__ {
 		sizex = <&ft5406>,"touchscreen-size-x:0";
 		sizey = <&ft5406>,"touchscreen-size-y:0";
--- a/arch/arm/boot/dts/overlays/vc4-kms-dsi-7inch-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-dsi-7inch-overlay.dts
@@ -113,6 +113,6 @@
 	};
 
 	__overrides__ {
-		disable_touch = <0>, "-10-11-12-13";
+		disable_touch = <0>, "-10-11-12";
 	};
 };
